src/transport/SecureSessionMgr.cpp

```diff
@@ -119,9 +121,29 @@ Transport::Type SecureSessionMgr::GetTransportType(NodeId peerNodeId)
 CHIP_ERROR SecureSessionMgr::SendMessage(SecureSessionHandle session, PayloadHeader & payloadHeader,
                                          System::PacketBufferHandle && msgBuf, EncryptedPacketBufferHandle * bufferRetainSlot)
 {
+    PeerConnectionState * state = nullptr;
     PacketHeader unusedPacketHeader;
-    return SendMessage(session, payloadHeader, unusedPacketHeader, std::move(msgBuf), bufferRetainSlot,
-                       EncryptionState::kPayloadIsUnencrypted);
+
+    state = GetPeerConnectionState(session);
+    VerifyOrReturnError(state != nullptr, CHIP_ERROR_NOT_CONNECTED);
+
+    if (!IsControlMessage(payloadHeader) && !state->GetSessionMessageCounter().GetPeerMessageCounter().IsSyncCompleted())
+    {
+        if (bufferRetainSlot != nullptr)
+        {
+            // If CRMP is enabled, skip queuing the message to avoid meltdown. (Check TCP meltdown for details)
```

Comment:
This comment should at the very least explain how the spec-defined semantics will end up happening in spite of the fact that we're not doing the spec-defined "queue the message" thing here.

Reply:
The message is already queued in the CRMP, here we can flush the CRMP queue once the counter is synced.

Reply:
Yes, I understand that.  I'm just saying we should document that this us what's happening.  And incidentally also do the flushing....